config/management/operational/src/validator_state.rs

```diff
@@ -2,67 +2,123 @@
 // SPDX-License-Identifier: Apache-2.0
 
 use crate::{json_rpc::JsonRpcClientWrapper, validator_config::DecryptedValidatorConfig};
-use diem_global_constants::{CONSENSUS_KEY, OWNER_ACCOUNT};
-use diem_management::error::Error;
+use diem_global_constants::{
+    CONSENSUS_KEY, FULLNODE_NETWORK_KEY, OWNER_ACCOUNT, VALIDATOR_NETWORK_KEY,
+};
+use diem_management::{
+    config::ConfigPath, error::Error, secure_backend::ValidatorBackend, storage::to_x25519,
+};
 use serde::Serialize;
 use structopt::StructOpt;
 
-#[derive(Debug, Default, Serialize)]
+#[derive(Debug, Serialize)]
 pub struct VerifyValidatorStateResult {
-    /// Check if a validator is in the latest validator set on-chain.
-    pub in_validator_set: Option<bool>,
     /// Check if the consensus key held in secure storage matches
     /// that registered on-chain for the validator.
     pub consensus_key_match: Option<bool>,
+
+    /// If the consensus_key_match is true, check if the key is unique
+    pub consensus_key_unique: Option<bool>,
+
+    /// Check if the fullnode network key held in secure storage matches
+    /// that registered on-chain.
+    pub fullnode_network_key_match: Option<bool>,
+
+    /// Check if a validator is in the latest validator set on-chain.
+    pub in_validator_set: Option<bool>,
+
+    /// Check if the validator network key held in secure storage matches
+    /// that registered on-chain.
+    pub validator_network_key_match: Option<bool>,
 }
 
 impl VerifyValidatorStateResult {
     pub fn is_state_consistent(&self) -> bool {
```

Comment:
nit: maybe rename to `is_valid_state`? I worry that `is_state_consistent` doesn't seem broad enough (e.g., to cover the case where we do uniqueness checks?)

Reply:
Changed to `is_valid_state`.